scripts/apps/search/directives/SaveSearch.ts

```diff
@@ -39,6 +39,13 @@ export function SaveSearch($location, asset, api, session, notify, $rootScope) {
                 scope.edit = null;
                 scope.activateSearchPane = false;
             };
+            scope.isValid = function(edit) {
+                if (edit.filter.query.raw == null) {
+                    return typeof edit.name === 'string' && edit.name.length > 0;
+                }
+                return edit.filter.query && typeof edit.filter.query.raw === 'string'
```

Comment:
imo it could use some helper func for the non empty string check, it's used 3x already